### ISSUE TITLE
Rule Change: `new-cap` should only exempt built-in names that reference the global

### ISSUE THREAD

**Issue by sethamus** [eslint/eslint#21268]
### What rule do you want to change?

`new-cap`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint new-cap: "error" */

function String(x) { return x; }

String(42);

const obj = {};

obj.String(42);
```

### What does the rule currently do for this code?

Nothing is reported, even though neither `String` is the global `String`.

### What will the rule do after it's changed?

Both calls are reported.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I agree that in both cases these are false-negatives, as such marking as accepted.
For the property case (`object.String`) global base objects (e.g. `globalThis` or `window`) should be exempted.

**Comment by namansancheti** [comment]
@sethamus
Are you planning to work on this ?
If not, I can pick this up - please let me know.
